### PR TITLE
feat: test module setup

### PR DESCRIPTION
**Pull request by qingzhuozhen**
### Summary
setup test module to host java sdk tessts
add a basic test
setup github action to run tests on pull request
add pull request template

### CODE REVIEW

File: src/demo/build.gradle [hunk 1]
```diff
@@ -29,7 +29,7 @@ sourceSets.main.java.srcDirs = ['java']
 dependencies {
     implementation 'org.json:json:20201115'
     implementation 'org.springframework.boot:spring-boot-starter-web'
-    compile project(':main')
+    implementation project(':main')
```
Comment (yuhao900914) [inline src/demo/build.gradle:32]: Good Catch. 
I just found the compile dependency configuration has been removed in the recently released Gradle 7.0.

File: .github/workflows/semantic-pr.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          !startsWith(github.event.pull_request.title, 'docs:') && !startsWith(github.event.pull_request.title, 'docs(') &&
+          !startsWith(github.event.pull_request.title, 'test:') && !startsWith(github.event.pull_request.title, 'test(') &&
+          !startsWith(github.event.pull_request.title, 'refactor:') && !startsWith(github.event.pull_request.title, 'refactor(') &&
+          !startsWith(github.event.pull_request.title, 'style:') && !startsWith(github.event.pull_request.title, 'style(') &&
+          !startsWith(github.event.pull_request.title, 'build:') && !startsWith(github.event.pull_request.title, 'build(') &&
+          !startsWith(github.event.pull_request.title, 'ci:') && !startsWith(github.event.pull_request.title, 'ci(') &&
+          !startsWith(github.event.pull_request.title, 'chore:') && !startsWith(github.event.pull_request.title, 'chore(') &&
+          !startsWith(github.event.pull_request.title, 'revert:') && !startsWith(github.event.pull_request.title, 'revert(')
+        run: |
+          echo 'Pull request title is not valid. Please check github.com/amplitude/Amplitude-JavaScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions'
```
Comment (yuhao900914) [inline .github/workflows/semantic-pr.yml:42]: Should we change the link to Amplitude-Java CONTRIBUTING.md?

File: src/demo/build.gradle (hunk 1 above)
Comment (qingzhuozhen) [reply src/demo/build.gradle:32]: yeah, need to update to get gradle run tests in action

File: .github/workflows/semantic-pr.yml (hunk 2 above)
Comment (qingzhuozhen) [reply .github/workflows/semantic-pr.yml]: right, copied it from iOS repo, just updated it and add that file in our repo.

**Review by yuhao900914 — APPROVED** [review]

**Pull request merged**